### PR TITLE
Consumer Groups

### PR DESCRIPTION
**Pull request by derekcollison**
Add in the ability for push consumers to optionally bind to a specific queue group.
The underlying sublist notifications also bind to the group as well and only report on interest that pertains to the queue group.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

**Review by kozlovic — CHANGES_REQUESTED** [review]
This is not doing entirely what we discussed in that the server is not letting the app know that you can't create a duplicate durable subscription, but also not rejecting AddConsumer with mismatch (queue/non_queue).
 
The only thing that the library can do with that is check from the Info that the DeliverGroup matches the user's `queue` value (either both empty, or mix and return appropriate errors). But I can't detect that the user is calling js.Subscribe() twice with a durable name that is currently active.

File: server/consumer.go [hunk 1]
```diff
@@ -43,14 +43,14 @@ type ConsumerInfo struct {
...
 	NumPending     uint64          `json:"num_pending"`
-	Active         *PushActive     `json:"active,omitempty"`
 	Cluster        *ClusterInfo    `json:"cluster,omitempty"`
 }
 
 type ConsumerConfig struct {
 	Durable         string        `json:"durable_name,omitempty"`
 	Description     string        `json:"description,omitempty"`
 	DeliverSubject  string        `json:"deliver_subject,omitempty"`
+	DeliverGroup    string        `json:"deliver_queue,omitempty"`
```
[the remark this answers is not in the document]
Comment (derekcollison) [reply server/consumer.go:53]: Will change

**Comment by derekcollison** [comment]
We discussed that I was removing push active and would add back in after this work if needed based on last delivered and last ack time stamps etc.

I also wanted to coordinate with you on the error and correct behavior, so we can do that today.

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**